include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,197 @@ namespace picongpu
+                    // end: write observer
+
+
+                    // start: write frequencies
+                    // prepare omega mesh
+                    ::openPMD::Mesh mesh_omega = openPMDdataFileIteration.meshes[dataLabelsDetectorFrequency(-1)];
+                    // write mesh attributes
+                    mesh_omega.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+                    mesh_omega.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    mesh_omega.setGridSpacing(std::vector<double>{1.0, 1.0, 1.0});
+                    mesh_omega.setGridGlobalOffset(std::vector<double>{0.0, 0.0, 0.0});
+                    mesh_omega.setGridUnitSI(1.0);
+                    mesh_omega.setAxisLabels(std::vector<std::string>{"None", "detector frequency index", "None"});
+                    std::map<::openPMD::UnitDimension, double> const unitDimensions_omega{
+                        {::openPMD::UnitDimension::T, -1.0}};
+                    mesh_omega.setUnitDimension(unitDimensions_omega);
+
+                    // write mesh attributes
+                    ::openPMD::MeshRecordComponent omega_mrc = mesh_omega[dataLabelsDetectorFrequency(0)];
                     const picongpu::float_64 factorOmega = 1.0 / UNIT_TIME;
-                    hdf5DataFile.writeAttribute(
-                        currentStep,
-                        radSplashType,
-                        (meshesPathName + dataLabelsDetectorFrequency(0)).c_str(),
-                        "unitSI",
-                        &factorOmega);
-
-                    /* position */
-                    std::vector<float_X> positionMesh(simDim, 0.0); /* there is no offset - zero */
-                    hdf5DataFile.writeAttribute(
-                        currentStep,
-                        splashFloatXType,
-                        (meshesPathName + dataLabelsDetectorFrequency(0)).c_str(),
-                        "position",
-                        1u,
-                        splash::Dimensions(simDim, 0, 0),
-                        &(*positionMesh.begin()));
+                    omega_mrc.setUnitSI(factorOmega);
+                    omega_mrc.setPosition(std::vector<double>{0.0, 0.0, 0.0});
 
+                    // reshape abstract MeshRecordComponent
+                    ::openPMD::Datatype datatype_omega
+                        = ::openPMD::determineDatatype(::openPMD::shareRaw(detectorFrequencies.data()));
+                    ::openPMD::Extent extent_omega = {1, radiation_frequencies::N_omega, 1};
+                    ::openPMD::Dataset dataset_omega = ::openPMD::Dataset(datatype_omega, extent_omega);
+                    omega_mrc.resetDataset(dataset_omega);
+
+                    // write actual data
+                    ::openPMD::Offset offset_omega = {0, 0, 0};
+                    omega_mrc.storeChunk(::openPMD::shareRaw(detectorFrequencies.data()), offset_omega, extent_omega);
```

Comment:
```suggestion
                    omega_mrc.storeChunk(detectorFrequencies.data(), offset_omega, extent_omega);
```
Same comments as above

Reply:
`.data()` has to be removed as well